core/templates/pages/community-dashboard-page/services/contribution-opportunities.service.ts

```diff
@@ -17,74 +17,149 @@
-        _getVoiceoverOpportunities(languageCode, '', successCallback);
-      },
-      getMoreSkillOpportunities: function(successCallback) {
-        if (moreSkillOpportunitiesAvailable) {
-          _getSkillOpportunities(skillOpportunitiesCursor, successCallback);
-        }
-      },
-      getMoreTranslationOpportunities: function(languageCode, successCallback) {
-        if (moreTranslationOpportunitiesAvailable) {
-          _getTranslationOpportunities(
-            languageCode, translationOpportunitiesCursor, successCallback);
-        }
-      },
-      getMoreVoiceoverOpportunities: function(languageCode, successCallback) {
-        if (moreVoiceoverOpportunitiesAvailable) {
-          _getVoiceoverOpportunities(
-            languageCode, voiceoverOpportunitiesCursor, successCallback);
-        }
-      }
-    };
-  }]);
+  }
+
+  private _getVoiceoverOpportunities(
+      languageCode: string,
+      cursor: string,
+      successCallback: (opportunities: Object, more: boolean) => void
+  ): void {
+    this.contributionOpportunitiesBackendApiService.fetchVoiceoverOpportunities(
+      languageCode, cursor).then((data: VoiceoverContributionOpportunities) => {
+      this.voiceoverOpportunitiesCursor = data.nextCursor;
+      this.moreVoiceoverOpportunitiesAvailable = data.more;
+      successCallback(data.opportunities, data.more);
+    });
+  }
+
+  getSkillOpportunities(
+      successCallback: (opportunities: Object, more: boolean) => void
+  ): void {
+    this._getSkillOpportunities('', successCallback);
+  }
+
+  getTranslationOpportunities(
+      languageCode: string,
+      successCallback: (opportunities: Object, more: boolean) => void
+  ): void {
+    this._getTranslationOpportunities(languageCode, '', successCallback);
+  }
+
+  getVoiceoverOpportunities(
+      languageCode: string,
+      successCallback: (opportunities: Object, more: boolean) => void
+  ): void {
+    this._getVoiceoverOpportunities(languageCode, '', successCallback);
+  }
+
+  getMoreSkillOpportunities(
+      successCallback: (opportunities: Object, more: boolean) => void
+  ): void {
+    if (this.moreSkillOpportunitiesAvailable) {
+      this._getSkillOpportunities(
+        this.skillOpportunitiesCursor, successCallback
+      );
+    }
+  }
+
+  getMoreTranslationOpportunities(
+      languageCode: string,
+      successCallback: (opportunities: Object, more: boolean) => void
+  ): void {
+    if (this.moreTranslationOpportunitiesAvailable) {
+      this._getTranslationOpportunities(
+        languageCode, this.translationOpportunitiesCursor, successCallback);
+    }
+  }
+
+  getMoreVoiceoverOpportunities(
+      languageCode: string,
+      successCallback: (opportunities: Object, more: boolean) => void
+  ): void {
+    if (this.moreVoiceoverOpportunitiesAvailable) {
+      this._getVoiceoverOpportunities(
+        languageCode, this.voiceoverOpportunitiesCursor, successCallback);
+    }
+  }
+}
```

Comment:
Can you change these functions to return promises? Wrap them inside a promise. You will have to change the instances where these functions are called as well.

Reply:
@srijanreddy98, I'm not sure how to do this. Can you help me with it?